pkg/migration/generate.go

```diff
@@ -516,6 +636,34 @@ func (h *Handler) addMonitoringSpec(cluster *corev1.StorageCluster) error {
 		prometheus.Spec.ServiceMonitorSelector != nil &&
 		prometheus.Spec.ServiceMonitorSelector.MatchLabels["name"] == serviceMonitorName {
 		cluster.Spec.Monitoring.Prometheus.Enabled = true
+		cluster.Status.DesiredImages.Prometheus = *prometheus.Spec.Image
+
+		dep := &appsv1.Deployment{}
+		err := h.client.Get(
+			context.TODO(),
+			types.NamespacedName{
+				Name:      component.PrometheusOperatorDeploymentName,
+				Namespace: cluster.Namespace,
+			},
+			dep,
+		)
+		if errors.IsNotFound(err) {
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		container := dep.Spec.Template.Spec.Containers[0]
+		cluster.Status.DesiredImages.PrometheusOperator = container.Image
+
+		for _, arg := range container.Args {
+			if strings.HasPrefix(arg, prometheusConfigReloaderArg) {
+				cluster.Status.DesiredImages.PrometheusConfigReloader = strings.TrimPrefix(arg, prometheusConfigReloaderArg)
```

Comment:
The existing customer have installed the prometheus operator using this  - https://docs.portworx.com/samples/k8s/portworx-pxc-operator.yaml. You can find this in the spec generator with DaemonSet on the final page.
This does not have the `--prometheus-config-reload` argument. If the arg is not passed prometheus operator uses the image with its own version. So after this for loop, let's check if the `DesiredImage.PrometheusConfigReloader` is empty. If empty, then let's add `quay.io/coreos/prometheus-config-reloader:<prometheus-image-version>` to the desired image.